pkgs/applications/editors/vscode/default.nix

```diff
@@ -53,9 +44,20 @@ in
       categories = "GNOME;GTK;Utility;TextEditor;Development;";
     };
 
-    buildInputs = if stdenv.hostPlatform.system == "x86_64-darwin"
-      then [ unzip libXScrnSaver libsecret ]
-      else [ wrapGAppsHook libXScrnSaver libxkbfile libsecret ];
+    buildInputs = (if stdenv.hostPlatform.system == "x86_64-darwin"
+      then [ unzip ]
+      else [ libxkbfile wrapGAppsHook ]) ++ [
+        nss
+        gtk2
+        nspr
+        gconf
+        alsaLib
+        libXtst
+        libsecret
+        at-spi2-atk
+        libXScrnSaver
+        autoPatchelfHook
```

Comment:
autoPatchelfHook is only available on linux. In fact only unzip would be useful on macOS.